hazelcast/src/main/java/com/hazelcast/spi/impl/proxyservice/impl/ProxyRegistry.java

```diff
@@ -91,16 +90,8 @@ boolean contains(String name) {
      * @param result The ProxyInfo of all proxies in this registry.
      */
     public void getProxyInfos(Collection<ProxyInfo> result) {
-        for (Map.Entry<String, DistributedObjectFuture> entry : proxies.entrySet()) {
-            final DistributedObjectFuture future = entry.getValue();
-            if (!future.isSet()) {
```

Comment:
Also after initializing of `future` we are sending events to inform all nodes. According to this, keeping `future.isSet()` check, may not cause much problems.